index/upsidedown/upsidedown.go

```diff
@@ -134,18 +134,23 @@ func (udc *UpsideDownCouch) loadSchema(kvreader store.KVReader) (err error) {
 	return
 }
 
+type RowBuffer struct {
+    buf *[]byte
+}
+
 var rowBufferPool sync.Pool
 
-func GetRowBuffer() []byte {
-	if rb, ok := rowBufferPool.Get().([]byte); ok {
-		return rb
-	} else {
-		return make([]byte, RowBufferSize)
-	}
+func GetRowBuffer() *RowBuffer {
+    	if rb, ok := rowBufferPool.Get().(*RowBuffer); ok {
+        	return rb
+    	} else {
+        	buf := make([]byte, RowBufferSize)
+        	return &RowBuffer{buf: &buf}
+    	}
 }
 
-func PutRowBuffer(buf []byte) {
-	rowBufferPool.Put(buf)
+func PutRowBuffer(rb *RowBuffer) {
```

Comment:
You've just changed the signature of these APIs without any changes to the callers!?! **This will break the build**.